rules/dispatch.go

```diff
@@ -387,3 +333,193 @@ func (r *DefaultRuleRenderer) dispatchChains(
 
 	return chains
 }
+
+func (r *DefaultRuleRenderer) dispatchEndPointMarkChains(
```

Comment:
If you make the above change, then you should change this to `endpointMarkDispatchChains` for consistency

Reply:
Yes.